src/library/AsyncWithdrawalLib.sol

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity ^0.8.24;
+
+import {VaultLib} from "src/library/VaultLib.sol";
+import {IVault} from "src/interface/IVault.sol";
+import {MainnetContracts as MC} from "script/Contracts.sol";
+import {IWithdrawalQueueManager} from "src/interface/IWithdrawalQueueManager.sol";
+import {IWithdrawalQueue} from "src/interface/external/lido/IWithdrawalQueue.sol";
+import {OriginWithdrawalLib} from "src/library/OriginWithdrawalLib.sol";
+
+library AsyncWithdrawalLib {
+    error UnsupportedAsset(address asset);
+
+    /**
+     * @notice Internal function to compute the total assets. Must handle the assets that are in queue for withdrawal.
+     * @dev This function should return the amount in base denomination.
+     */
+    function computeTotalAssets() public view returns (uint256 totalBaseBalance) {
+        totalBaseBalance = VaultLib.computeTotalAssets();
+
+        // Get storage
+        IVault.AssetStorage storage assetStorage = VaultLib.getAssetStorage();
+
+        // Iterate through assets
+        address[] memory assetList = assetStorage.list;
+        uint256 assetListLength = assetList.length;
+
+        for (uint256 i = 0; i < assetListLength; i++) {
+            totalBaseBalance += asyncWithdrawalBalance(assetList[i]);
+        }
+    }
+
+    /**
+     * @notice function to handle the assets that are in queue for withdrawal.
+     * @param asset_ The address of the asset.
+     * @dev This function should return the amount in base denomination.
+     */
+    function asyncWithdrawalBalance(address asset_) public view returns (uint256 baseAssets) {
+        baseAssets = _asyncWithdrawalBalance(asset_);
+    }
+
+    function _asyncWithdrawalBalanceYNAsset(address queueManager_) private view returns (uint256 baseAssets) {
+        IWithdrawalQueueManager queueManager = IWithdrawalQueueManager(queueManager_);
+        (, IWithdrawalQueueManager.WithdrawalRequest[] memory requests) =
+            queueManager.withdrawalRequestsForOwner(address(this));
+        for (uint256 i = 0; i < requests.length; i++) {
+            if (!requests[i].processed) {
+                // NOTE: needs to be fixed - assumes no slashing for now, as in reality eigenlayer slashing is not active yet
+                // also we do not account for the fees here
+                baseAssets += requests[i].amount * requests[i].redemptionRateAtRequestTime / 1e18;
+                // subtract fee
+                baseAssets -= queueManager.calculateFee(requests[i].amount, requests[i].feeAtRequestTime);
```

Comment:
this is incorrect as it substracts the fee in the unit of measure of amount, which is ynETH or ynLSDe

while basAssets is already converted

apply the fee first and convert after